dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -152,4 +156,36 @@ void PushingToViewsBlockOutputStream::flush()
         view.out->flush();
 }
 
+void PushingToViewsBlockOutputStream::process(const Block & block, size_t view_num)
+{
+    auto & view = views[view_num];
+
+    try
+    {
+        BlockInputStreamPtr from = std::make_shared<OneBlockInputStream>(block);
+        InterpreterSelectQuery select(view.query, *views_context, from);
+        BlockInputStreamPtr in = std::make_shared<MaterializingBlockInputStream>(select.execute().in);
+        /// Squashing is needed here because the materialized view query can generate a lot of blocks
+        /// even when only one block is inserted into the parent table (e.g. if the query is a GROUP BY
+        /// and two-level aggregation is triggered).
+        in = std::make_shared<SquashingBlockInputStream>(
+            in, context.getSettingsRef().min_insert_block_size_rows, context.getSettingsRef().min_insert_block_size_bytes);
+
+        in->readPrefix();
+
+        while (Block result_block = in->read())
+        {
+            Nested::validateArraySizes(result_block);
+            view.out->write(result_block);
+        }
+
+        in->readSuffix();
+    }
+    catch (Exception & ex)
+    {
+        ex.addMessage("while pushing to view " + view.database + "." + view.table);
+        exceptions[view_num] = std::current_exception();
+    }
```

Comment:
This will lead to std::terminate in case of non DB::Exception.
Examples: bad alloc, network error, etc.

Reply:
The issue will be solved automatically if you use `ThreadPool`.